core/src/main/scala/org/apache/spark/storage/BlockManagerManagedBuffer.scala

```diff
@@ -31,17 +35,31 @@ import org.apache.spark.util.io.ChunkedByteBuffer
 private[storage] class BlockManagerManagedBuffer(
     blockInfoManager: BlockInfoManager,
     blockId: BlockId,
-    chunkedBuffer: ChunkedByteBuffer) extends NettyManagedBuffer(chunkedBuffer.toNetty) {
+    data: BlockData,
+    dispose: Boolean) extends ManagedBuffer {
+
+  private val refCount = new AtomicInteger(1)
```

Comment:
maybe we should mention it in the class doc that the `BlockData` will be disposed automatically via reference count.